### PR TITLE
Issue 60940

### PR DESCRIPTION
**Pull request by rwhundley**
[Originating issue](https://github.ibm.com/ibmprivatecloud/roadmap/issues/60940)

Reconcile logic was missing returns for errors found during creates or
updates of Client CRs, which led to failing CRs never reconciling
properly unless retriggered with a small update.

Other changes:

* Remove vestigial CACHE_UPDATE check

Signed-off-by: Rob Hundley <rwhundle@us.ibm.com>

### CODE REVIEW

File: pkg/controller/client/client_controller.go [hunk 1]
```diff
@@ -333,14 +329,22 @@ func (r *ReconcileClient) processOidcRegistration(ctx context.Context, client *o
...
 		err = r.createClient(ctx, client)
 	} else {
 		reqLogger.Info("Client found, update Client")
+		verb = "update"
 		err = r.updateClient(ctx, client)
 	}
 
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Error occured while attempting to %s Client", verb))
+		return
```
Comment (rashmi43) [inline pkg/controller/client/client_controller.go:345]: should we return the error as well?

**Review by rashmi43 — APPROVED** [review]

File: pkg/controller/client/client_controller.go (hunk 1 above)
Comment (rwhundley) [reply pkg/controller/client/client_controller.go:345]: This is a naked return from a function with named returns, which means that `return` is the same as `return err` in this context.

Comment (rwhundley) [reply pkg/controller/client/client_controller.go:345]: Example of how this is used here: [A Tour of Go](https://go.dev/tour/basics/7)

**Review by rashmi43 — APPROVED** [review]

File: pkg/controller/client/client_controller.go (hunk 1 above)
Comment (rashmi43) [reply pkg/controller/client/client_controller.go:345]: ok

**Pull request merged**